troposphere/static/js/components/modals/instance/InstanceLaunchWizardModal.react.js

```diff
@@ -433,16 +330,21 @@ export default React.createClass({
     },
 
     onSubmitLaunch: function() {
-        let basic = this.state.basicLaunch;
         let scripts = this.state.advancedOptions
             .bootScriptOption.attachedScripts;
 
+        // TODO: this object is the real state for this component (it's what's
+        // necessary) this entire method should be simplified to the following:
+        //
+        // actions.InstanceActions.launch(this.state);
+        // this.hide();
+
         let launchData = {
-            project: basic.project,
-            instanceName: basic.instanceName,
-            identity: basic.identityProvider,
-            size: basic.providerSize,
-            version: basic.imageVersion,
+            project: this.state.project,
+            instanceName: this.state.instanceName || this.state.imageName,
```

Comment:
`this.state.instanceName || this.state.imageName` this is key. the instanceName state is separate from the imageName, this makes the logic easy, in other places. the only thing that sets the instanceName ever is when someone types in the input